content/java/concepts/compiler/compiler.md

```diff
@@ -0,0 +1,37 @@
+---
+Title: 'Compiler'
+Description: 'Java compilers typically take Java source code and produce class files containing platform-neutral Java bytecode that can be executed by the Java Virtual Machine (JVM).'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Command Line'
+  - 'Developer Tools'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+Java **compilers** are programs that take Java source code and produce class files containing platform-neutral Java bytecode that can be executed by the Java Virtual Machine (JVM).
+
+The JVM does not interpret source code. Instead, it interprets what is called Java bytecode (a language somewhere between human-readable source code, and [machine code](https://www.codecademy.com/resources/docs/general/machine-code)), which is specific to a particular computer. Unlike machine code, bytecode is platform-neutral meaning the same bytecode can be interpreted by any JVM running on any sort of computer system. This is what makes compiled Java programs portable.
```

Comment:
```suggestion
The JVM does not interpret source code (high-level program). Instead, it interprets what is called Java bytecode (a low-level language somewhere between human-readable source code, and [machine code](https://www.codecademy.com/resources/docs/general/machine-code)), which is specific to a particular computer. Unlike machine code, bytecode is platform-neutral meaning the same bytecode can be interpreted by any JVM running on any sort of computer system. This is what makes compiled Java programs portable.
```

Reply:
Between the first and second sentence, can we explain how the compiler converts the source code into the low-level program? The Java byte code is the object file or result of the compiled source code, or something along those lines just to be more explicit about the high-to-low language conversion

Reply:
It's mentioned in the prior paragraph that the compiler produces the bytecode. I think adding something here would be redundant.